nixos/modules/services/databases/neo4j.nix

```diff
@@ -5,28 +5,81 @@ with lib;
 let
   cfg = config.services.neo4j;
 
+  sslPolicies = mapAttrsToList (
+    name: conf: ''
+      dbms.ssl.policy.${name}.allow_key_generation=${boolToString conf.allowKeyGeneration}
+      dbms.ssl.policy.${name}.base_directory=${builtins.replaceStrings ["__policyname__"] [name] (toString conf.baseDirectory)}
+      ${optionalString (conf.ciphers != null) ''
+        dbms.ssl.policy.${name}.ciphers=${concatStringsSep "," conf.ciphers}
+      ''}
+      dbms.ssl.policy.${name}.client_auth=${conf.clientAuth}
+      ${if builtins.length (splitString "/" conf.privateKey) > 1 then
+        ''dbms.ssl.policy.${name}.private_key=${conf.privateKey}''
+      else
+        ''dbms.ssl.policy.${name}.private_key=${builtins.replaceStrings ["__policyname__"] [name] (toString conf.baseDirectory)}/${conf.privateKey}''
+      }
+      ${if builtins.length (splitString "/" conf.privateKey) > 1 then
+        ''dbms.ssl.policy.${name}.public_certificate=${conf.publicCertificate}''
+      else
+        ''dbms.ssl.policy.${name}.public_certificate=${builtins.replaceStrings ["__policyname__"] [name] (toString conf.baseDirectory)}/${conf.publicCertificate}''
+      }
+      dbms.ssl.policy.${name}.revoked_dir=${builtins.replaceStrings ["__policyname__"] [name] (toString conf.revokedDir)}
+      dbms.ssl.policy.${name}.tls_versions=${concatStringsSep "," conf.tlsVersions}
+      dbms.ssl.policy.${name}.trust_all=${boolToString conf.trustAll}
+      dbms.ssl.policy.${name}.trusted_dir=${builtins.replaceStrings ["__policyname__"] [name] (toString conf.trustedDir)}
```

Comment:
This `__policyname__` replacement business isn't really necessary, because in the end you're gonna have the name at the definition anyways:

```nix
services.neo4j.policies.foo = {
  trustedDir = "/some/foo";
}
```

I'll comment below how you can use the policy name in the options default value without having to use this replacement.

Reply:
I figured there was a way but couldn't pinpoint it..